tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,261 @@
+package charts
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/charts"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	// Project that charts are installed in
+	cisbenchmarkProjectName = "cis-operator-system"
+	clusterScanResourceType = "cis.cattle.io.clusterscan"
+	clusterScanReportType   = "cis.cattle.io.clusterscanreport"
+	CisConfigFileKey        = "cis"
+	ClusterScanYaml         = `
+apiVersion: cis.cattle.io/v1
+kind: ClusterScan
+metadata:
+  name: %s #string
+  annotations:
+    {}
+#    key: string
+spec:
+  scanProfileName: %s
+  scheduledScanConfig:
+    retentionCount: 3
+    scanAlertRule:
+#      alertOnComplete: boolean
+#      alertOnFailure: boolean
+#    cronSchedule: string
+  scoreWarning: pass
+  `
+)
+
+type CisConfig struct {
+	ProfileName string `json:"profileName" yaml:"profileName"`
+}
+
+type ClusterScanStatus struct {
+	Display Display
+}
+
+type Display struct {
+	State string
+}
+
+type ClusterScanReportSpec struct {
+	ReportJson string
+}
+
+type CisReport struct {
+	Total         int
+	Pass          int
+	Fail          int
+	Skip          int
+	Warn          int
+	NotApplicable int
+	Results       []*Group `json:"results"`
+}
+
+type Group struct {
+	ID     string      `yaml:"id" json:"id"`
+	Text   string      `json:"description"`
+	Checks []*CisCheck `json:"checks"`
+}
+
+type CisCheck struct {
+	Id          string
+	Description string
+	State       string
+}
```

Comment:
I think these can be either in `cisbenchmark.go` in the same dir or in `extensions/charts` depending on how often they'd be used.

Reply:
Also, a small nit, since these are public, they need to be explained for godoc as comments

Reply:
I think they need to be extensions especially if they are public imo